### PR TITLE
Added Github Action for deploying Docker image to Docker Hub

### PR DESCRIPTION
**Pull request by Flurb**
Some questions are unanswered:
- The Github Action is now triggered on every push. For the develop branch, this is probably fine (keep publishing a snapshot), but for a Master / release branch it's preferred to trigger on release like this:
```yaml
on:
  release:
    types: [published]
```
This needs to be discussed first, also see the issue about release management in the compas-architecture repository.
More info: https://dev.to/ivandotv/automate-publishing-of-your-github-repository-to-docker-hub-5872.
- A private Docker Hub account is used for storing the image. According to the LF Energy community, this is okay but I don't agree. We need a better solution for this. Until then, we can use this one.

### CODE REVIEW

File: .github/workflows/dockerhub_deployment.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  push_to_registry:
+    name: Build and publish
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+      - name: Login to Docker Hub
+        run: echo ${{ secrets.DOCKER_PASSWORD }} | docker login -u ${{ secrets.DOCKER_USERNAME }} --password-stdin
+      - name: Build and publish Docker Image
+        run: ./gradlew clean build -Pnative -Dquarkus.native.container-build=true -Dquarkus.container-image.build=true -Dquarkus.container-image.group=lfenergycompas -Dquarkus.container-image.name=scl-data-service -Dquarkus.container-image.push=true
```
Comment (dlabordus) [inline .github/workflows/dockerhub_deployment.yml:19]: Why not start the name of the container also with compas?
-Dquarkus.container-image.name=compas-scl-data-service

That may be better if someone searches for containers.

File: .github/workflows/dockerhub_deployment.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# SPDX-FileCopyrightText: 2021 Alliander N.V.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: Docker Hub Deployment
+
+on: push
```
Comment (dlabordus) [inline .github/workflows/dockerhub_deployment.yml:7]: Should there be a filter on "develop" and "main"?
At least not on all the feature branches.

And maybe just go for the "release" event as described in your comment. Mention the solution during the refinement and we will see in the future if this is the way to go.

File: app/build.gradle [hunk 3]
```diff
@@ -9,6 +9,7 @@ plugins {
 }
 
 dependencies {
+    implementation 'io.quarkus:quarkus-container-image-docker'
```
Comment (dlabordus) [inline app/build.gradle:12]: Why not move it above the current line 15.
It looks nicer with the other Quarkus implementations.
And the BOM is then always the first one.

File: .github/workflows/dockerhub_deployment.yml (hunk 2 above)
Comment (Flurb) [reply .github/workflows/dockerhub_deployment.yml]: I removed the on push, and replaced it with the following:

```yaml
on:
  release:
    types: [released]
```

According to https://docs.github.com/en/actions/reference/events-that-trigger-workflows, it should now be triggered on when a release is 'released'.

**Review by dlabordus — DISMISSED** [review]

**Review by dlabordus — APPROVED** [review]

**Pull request merged**